ui/src/app/icons/edit/component.tsx

```diff
@@ -0,0 +1,77 @@
+import {Component} from '@angular/core';
+
+import * as React from 'react';
+
+import {ReactWrapperBase} from 'app/utils';
+
+interface EditComponentState { style: object; }
+
+const style = {
+  height: 14,
+  width: 14,
+};
+
+const fillStyle = {
+  fill: '#4D4CA5'
+};
+
+class EditComponentReact extends React.Component<{}, EditComponentState> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      style: style
+    };
+  }
+
+  mouseOver(): void {
+    this.setState({style: {...style, ...fillStyle}});
+  }
+
+  mouseLeave(): void {
+    this.setState({style: style});
+  }
+
+  render() {
+    return (
+      <svg style={this.state.style}
+           viewBox='0 0 14 14'
+           version='1.1'
+           xmlns='http://www.w3.org/2000/svg'
+           xmlnsXlink='http://www.w3.org/1999/xlink'
+           onMouseOver={() => this.mouseOver()}
```

Comment:
FYI, I'm about to add `react-interactive` which will allow us to add hover styles much more easily, without tracking the state by hand. So don't fret too much about the implementation of this.